fixtures-processor/src/main/kotlin/com/theblueground/fixtures/ParameterValueGenerator.kt

```diff
@@ -233,21 +236,27 @@ internal class ParameterValueGenerator {
     }
 
     private fun generateSealedValue(
-        randomize: Boolean,
+        kspArguments: KspArguments,
         parameter: ProcessedFixtureParameter.SealedParameter,
     ): String {
-        val sealedEntry = if (randomize) {
-            parameter.entries.random()
+        val entries = parameter.entries.filter { it.isObject || it.isFixture }
+        require(entries.isNotEmpty()) {
+            "At least one sealed data class annotated with @Fixture is required in: ${parameter.type.copy(nullable = false)}"
+        }
+        val sealedEntry = if (kspArguments.randomize) {
+            entries.random()
         } else {
-            parameter.entries.first()
+            entries.first()
         }
 
         return when {
             sealedEntry.isObject -> "${parameter.typeName}.${sealedEntry.name}"
-            sealedEntry.isFixture -> "create${parameter.typeName}${sealedEntry.name}()"
-            else -> throw IllegalArgumentException(
-                "Sealed data classes that are fields of a Fixture should be annotated with @Fixture too",
-            )
+            sealedEntry.isFixture -> {
+                val functionName = "${kspArguments.prefix}${parameter.typeName}${sealedEntry.name}".replaceFirstChar { it.lowercaseChar() }
+                "$functionName()"
+            }
+
+            else -> error("${parameter.typeName}.${sealedEntry.name} must be an object or fixture based on filter in preconditions")
```

Comment:
👏 This is very helpful for understanding on which class the generation failed. Great addition!